core/common/src/files/FileSystem.kt

```diff
@@ -129,6 +129,19 @@ public sealed interface FileSystem {
      * @param path the path to get the metadata for.
      */
     public fun metadataOrNull(path: Path): FileMetadata?
+
+    /**
+     * Returns an absolute path to the same file or directory the [path] is pointing to.
+     * All symbolic links are solved, extra path separators and references to current (`.`) or
+     * parent (`..`) directories are removed.
+     * If the [path] is a relative path then it'll be resolved against current working directory.
+     * If there is no file or directory to which the [path] is pointing to then [FileNotFoundException] will be thrown.
+     *
+     * @param path the path to resolve.
+     * @return a resolved path.
+     * @throws FileNotFoundException if there is no file or directory corresponding to the specified path.
```

Comment:
Is there a good reason for this behavior? It seems that our file API has functions that create a file by the passed `Path`, like `createDirectories`, `atomicMove`, `FileSystem#sink` - in this case, it will be impossible to use `resolve`.

Reply:
The `realpath` function (to which native targets delegate resolution) works only for paths corresponding to existing files/directories. That's the main reason.

Reply:
:(